src/Microsoft.AspNetCore.Routing.Abstractions/RouteValueDictionary.cs

```diff
@@ -269,8 +314,30 @@ IEnumerator IEnumerable.GetEnumerator()
         /// <inheritdoc />
         bool ICollection<KeyValuePair<string, object>>.Remove(KeyValuePair<string, object> item)
         {
-            EnsureWritable();
-            return ((ICollection<KeyValuePair<string, object>>)InnerDictionary).Remove(item);
+            if (_storage.Count == 0)
+            {
+                return false;
+            }
+
+            if (_storage.Count == 0)
```

Comment:
Like the new `Count` checks. But I suspect you'll find the value hasn't changed since [line 317](https://github.com/aspnet/Routing/pull/309/files#diff-8a515f05675a880115920385ab225fcbR317) :trollface: